crates/biome_css_analyze/src/lint/nursery/no_duplicate_selectors.rs

```diff
@@ -0,0 +1,327 @@
+use std::borrow::Borrow;
+use std::collections::HashSet;
+use std::hash::{DefaultHasher, Hash, Hasher};
+use std::vec;
+
+use biome_analyze::Ast;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{
+    AnyCssAtRule, AnyCssRelativeSelector, AnyCssRule, AnyCssSelector, CssComplexSelector,
+    CssRelativeSelector, CssRelativeSelectorList, CssRoot, CssSelectorList, CssSyntaxNode,
+};
+use biome_deserialize_macros::Deserializable;
+use biome_rowan::{AstNode, SyntaxNodeCast};
+
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Disallow duplicate selectors.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// .abc,
+    /// .def,
+    /// .abc {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```
+    /// .foo {}
+    /// .bar {}
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// If true, disallow duplicate selectors within selector lists.
+    ///
+    /// ```json
+    /// {
+    ///     "noDuplicateSelectors": {
+    ///         "options": {
+    ///           "disallowInList": true
+    ///         }
+    ///     }
+    /// }
+    /// ```
```

Comment:
```suggestion
    /// ```json5, ignore
    /// {
    ///	// ...
    ///     "noDuplicateSelectors": {
    ///         "options": {
    ///           "disallowInList": true
    ///         }
    ///     }
    ///	// ...
    /// }
    /// ```
```